src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -62,15 +67,16 @@ public BookDto updateBookById(Long id, BookRequestDto updatedBookDto) {
     }
 
     @Override
-    public void deleteById(Long id) {
-        bookRepository.deleteById(id);
-    }
-
-    @Override
-    public List<BookDto> search(BookSearchParameters parameters) {
+    public List<BookDto> search(BookSearchParameters parameters,Pageable pageable) {
         Specification<Book> bookSpecification = bookSpecificationBuilder.build(parameters);
         return bookRepository.findAll(bookSpecification).stream()
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
```

Comment:
I think I have already seen that you have moved this method down in one of your previous tasks, maybe it is not you doing it by hand, but in case you did - there is no point in doing the same thing in multiple branches if you know that one of them is going to main, and you can get merge conflicts because of that